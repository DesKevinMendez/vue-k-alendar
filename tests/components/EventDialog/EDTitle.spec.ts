import EDTitle from '@/components/EventDialog/EDTitle.vue';
import type { KEvent } from '@/types';
import { mount, VueWrapper } from '@vue/test-utils';
import { beforeEach, describe, expect, it, vi } from 'vitest';

let wrapper: VueWrapper<InstanceType<typeof EDTitle>>

const event: KEvent = {
  title: 'Event 1',
  start_date: '2024-10-13',
  description: 'Description of random event',
  color: '#fe36d9',
  autor: 'Kevin MÃ©ndez',
  id: 'ffee34f5-5ebb-487d-ab21-61f42edbc1df',
}

describe('EDTitle', () => {

  beforeEach(() => {
    wrapper = mount(EDTitle, {
      props: {
        event,
      }
    });
  })

  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('snapshot', () => {
    beforeEach(() => {
      vi.clearAllMocks();
    });

    it('should render the component', async () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('event is null', async () => {
      await wrapper.setProps({ event: null });
      expect(wrapper.html()).toMatchSnapshot();
      const span = wrapper.find('span');

      expect(span.exists()).toBeFalsy();
    })
  })

  it('span circle', () => {
    const span = wrapper.find('span');
    expect(span.exists()).toBeTruthy();
    expect(span.classes()).toContain('circle');
    expect(span.attributes().style).toBe('background-color: rgb(254, 54, 217);');
  })
});