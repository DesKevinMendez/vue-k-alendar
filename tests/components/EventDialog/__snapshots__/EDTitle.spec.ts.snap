// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EDTitle > should render the component 1`] = `
"<div data-v-3f995ea3="" class="title"><span data-v-3f995ea3="" class="circle" style="background-color: rgb(254, 54, 217);"></span>
  <h2 data-v-3f995ea3="">Event 1</h2>
</div>"
`;

exports[`EDTitle > snapshot > event is null 1`] = `
"<div data-v-3f995ea3="" class="title">
  <!--v-if-->
  <h2 data-v-3f995ea3=""></h2>
</div>"
`;

exports[`EDTitle > snapshot > should render the component 1`] = `
"<div data-v-3f995ea3="" class="title"><span data-v-3f995ea3="" class="circle" style="background-color: rgb(254, 54, 217);"></span>
  <h2 data-v-3f995ea3="">Event 1</h2>
</div>"
`;
